services/ymax-planner/src/engine.ts

```diff
@@ -547,36 +619,45 @@ export const startEngine = async (
     }
   }).done;
 
-  const txPowers = {
+  const txPowers: HandlePendingTxOpts = Object.freeze({
     ...evmCtx,
     signingSmartWalletKit,
     fetch,
     cosmosRest,
-    marshaller,
     log: console.warn.bind(console),
-    error: console.error.bind(console),
-  };
+  });
   console.warn(`Found ${pendingTxKeys.length} pending transactions`);
-  await makeWorkPool(pendingTxKeys, undefined, async txId => {
-    const path = `${PENDING_TX_PATH_PREFIX}.${txId}`;
-    const errLabel = `🚨 Failed to process old pending tx ${path}`;
 
+  const initialPendingTxData: PendingTxRecord[] = [];
+  await makeWorkPool(pendingTxKeys, undefined, async (txId: TxId) => {
+    const path = `${PENDING_TX_PATH_PREFIX}.${txId}`;
     await null;
-    let data;
+    let streamCellJson;
+    let data: unknown;
     try {
-      data = await query.readPublished(stripPrefix('published.', path));
+      streamCellJson = await query.vstorage.readStorage(path, {
+        kind: 'data',
+      });
+      const streamCell = parseStreamCell(streamCellJson, path);
+      data = parseStreamCellValue(streamCell, -1, path);
       mustMatch(data, PublishedTxShape, path);
-      const tx = { txId, ...data } as PendingTx;
-      console.warn('Old pending tx', tx);
-      // Tx resolution is non-blocking.
-      void handlePendingTx(tx, txPowers).catch(err =>
-        console.error(errLabel, err),
-      );
+      initialPendingTxData.push({
+        blockHeight: BigInt(streamCell.blockHeight),
+        tx: { txId, ...data },
+      });
     } catch (err) {
-      console.error(errLabel, data, err);
+      const errLabel = ` Failed to process old pending tx ${path}`;
+      console.error(errLabel, data || streamCellJson, err);
     }
   }).done;
 
+  await processInitialPendingTransactions(
+    initialPendingTxData,
+    rpc,
+    txPowers,
+    now,
+  );
```

Comment:
`rpc` and `now` should be included in `txPowers`.
```suggestion
  await processInitialPendingTransactions(initialPendingTxData, txPowers);
```